platforms/documentation/docs/src/docs/release/notes.md

```diff
@@ -71,6 +71,34 @@ Starting with Gradle 9.0, the Gradle API is annotated using JSpecify instead.
 
 For more details and potential breakages, see the dedicated [upgrading guide section](userguide/upgrading_version_8.html).
 
+### Task graph diagnostic
+
+We introduce a new task graph diagnostic feature to help you understand the task graph of your build.
+You can use `--task-graph` command line option to print the task graph of your build without executing the tasks, for example:
+```
+$ ./gradlew root r2 --task-graph
+
+Tasks graph for: root r2
++--- :root (org.gradle.api.DefaultTask)
+|    \\--- :middle (org.gradle.api.DefaultTask)
+|         +--- :leaf1 (org.gradle.api.DefaultTask)
+|         \\--- :leaf2 (org.gradle.api.DefaultTask, disabled)
+\\--- :root2 (org.gradle.api.DefaultTask)
+    +--- :leaf1 (org.gradle.api.DefaultTask) (*)
+    |--- other build task :included:fromIncluded (org.gradle.api.DefaultTask)
+    \\--- :leaf4 (org.gradle.api.DefaultTask, finalizer)
+         \\--- :leaf3 (org.gradle.api.DefaultTask)
+         
+(*) - details omitted (listed previously)
+```
+
+This feature can be useful to get a quick overview of the task graph and to understand the dependencies between tasks.
```

Comment:
Feel free to change this but it would be nice as a list:
```suggestion
This feature is especially useful for:
- Gaining a quick overview of task dependencies
- Debugging why a task is scheduled
- Exploring subgraphs by refining your task selection
```

Reply:
I think these items are from different categories, so it feels a bit weird to me to put them on one list.

Reply:
Let's do this instead:
```suggestion
This feature provides a quick overview of the task graph, helping users understand the dependencies between tasks without running them.
```